patches/minecraft/net/minecraft/advancements/AdvancementRewards.java.patch

```diff
@@ -1,5 +1,28 @@
 --- a/net/minecraft/advancements/AdvancementRewards.java
 +++ b/net/minecraft/advancements/AdvancementRewards.java
+@@ -29,12 +_,18 @@
+    private final ResourceLocation[] f_9980_;
+    private final ResourceLocation[] f_9981_;
+    private final CommandFunction.CacheableFunction f_9982_;
++   private final net.minecraftforge.common.advancements.IAdvancementReward[] rewards;
++
++   public AdvancementRewards(int experience, ResourceLocation[] loot, ResourceLocation[] recipes, CommandFunction.CacheableFunction function, net.minecraftforge.common.advancements.IAdvancementReward[] rewards) {
++      this.f_9979_ = experience;
++      this.f_9980_ = loot;
++      this.f_9981_ = recipes;
++      this.f_9982_ = function;
++      this.rewards = rewards;
++   }
+ 
+    public AdvancementRewards(int p_9985_, ResourceLocation[] p_9986_, ResourceLocation[] p_9987_, CommandFunction.CacheableFunction p_9988_) {
+-      this.f_9979_ = p_9985_;
+-      this.f_9980_ = p_9986_;
+-      this.f_9981_ = p_9987_;
+-      this.f_9982_ = p_9988_;
++      this(p_9985_, p_9986_, p_9987_, p_9988_, new net.minecraftforge.common.advancements.IAdvancementReward[0]);
+    }
```

Comment:
I meant make this entire patch be what I showed, not make the patch even bigger... The line below the existing constructor's signature you add a this call, then you end the constructor, then you start the new constructor that you are adding. Use srg names in it for all but the parameter you are adding and then add the new constructor's signature to the exc file so that the params can be properly mapped.

Reply:
I have been told that signature change in patches is not wanted

Reply:
I never said to change the signature?

Reply:
Then I don't understand what you want me to change

Reply:
```java
//Existing constructor
public AdvancementRewards(int p_9985_, ResourceLocation[] p_9986_, ResourceLocation[] p_9987_, CommandFunction.CacheableFunction p_9988_) {
//added code
   this(p_9985_, p_9986_, p_9987_, p_9988_, new net.minecraftforge.common.advancements.IAdvancementReward[0]);
}
public AdvancementRewards(int p_9985_, ResourceLocation[] p_9986_, ResourceLocation[] p_9987_, CommandFunction.CacheableFunction p_9988_, net.minecraftforge.common.advancements.IAdvancementReward... rewards) {
//Existing constructor body
//line you are adding to the body
this.rewards = rewards;
```